habitat-lab/habitat/tasks/rearrange/actions/oracle_nav_action.py

```diff
@@ -107,47 +129,97 @@ def step(self, *args, is_last_action, **kwargs):
         final_nav_targ, obj_targ_pos = self._get_target_for_idx(
             nav_to_target_idx
         )
-        cur_nav_targ = self._path_to_point(final_nav_targ)[1]
 
-        robot_pos = np.array(self.cur_articulated_agent.base_pos)
         base_T = self.cur_articulated_agent.base_transformation
-        forward = np.array([1.0, 0, 0])
-        robot_forward = np.array(base_T.transform_vector(forward))
-
-        # Compute relative target.
-        rel_targ = cur_nav_targ - robot_pos
-        # Compute heading angle (2D calculation)
-        robot_forward = robot_forward[[0, 2]]
-        rel_targ = rel_targ[[0, 2]]
-        rel_pos = (obj_targ_pos - robot_pos)[[0, 2]]
-
-        angle_to_target = get_angle(robot_forward, rel_targ)
-        angle_to_obj = get_angle(robot_forward, rel_pos)
+        if self.motion_type != "base_velocity":
```

Comment:
The `step` function is now getting a too long and complicated. Can you break this up into sub-methods to make it clear what the individual components are doing?

Reply:
I did some changes that reduce it. Let me know if that is still too long.